prime-router/src/main/kotlin/azure/ApiKeysFunctions.kt

```diff
@@ -15,10 +15,78 @@ import gov.cdc.prime.router.tokens.AuthenticatedClaims
+import io.swagger.v3.oas.annotations.OpenAPIDefinition
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.enums.SecuritySchemeType
+import io.swagger.v3.oas.annotations.info.Contact
+import io.swagger.v3.oas.annotations.info.Info
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.OAuthFlow
+import io.swagger.v3.oas.annotations.security.OAuthFlows
+import io.swagger.v3.oas.annotations.security.OAuthScope
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import io.swagger.v3.oas.annotations.security.SecurityScheme
 import org.apache.logging.log4j.kotlin.Logging
+import javax.ws.rs.DELETE
+import javax.ws.rs.GET
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
 
-class ApiKeysFunctions(private val settingsFacade: SettingsFacade = SettingsFacade.common) : Logging {
+const val KEY_MGMT_TAG = "Public Key Management"
+const val HTTP_401_ERR_MSG = "Unauthorized operation"
+const val HTTP_404_ERR_MSG = "API key not found"
+const val HTTP_400_ERR_MSG = "Bad request"
+const val HTTP_200_GET_MSG = "API keys returned"
+const val HTTP_200_DELETE_MSG = "API key deleted"
+const val HTTP_200_POST_MSG = "API key created"
+const val HTTP_404_DELETE_MSG = "Organization, scope, or key id not found"
+
+const val PRIME_ADMIN_PATTERN = "*.*.primeadmin"
+const val ORG_NAME_DESC = "the organization whose keys are retrieved"
+const val OPERATION_GET_KEYS_DESC = "Retrieve API key(s) for the given organization"
+const val ORG_NAME_STR = "organizationName"
+const val SCOPE_STR = "scope"
+const val KID_STR = "kid"
+const val HTTP_200_OK = "200"
+const val HTTP_400_BAD_REQ = "400"
+const val HTTP_401_UNAUTHORIZED = "401"
+const val HTTP_404_NOT_FOUND = "404"
 
+@OpenAPIDefinition(
+    info = Info(
+        title = "Prime ReportStream",
+        description = "A router of public health data from senders to receivers",
+        contact = Contact(
+            name = "USDS at Centers for Disease Control and Prevention",
+            url = "https://reportstream.cdc.gov",
+            email = "reportstream@cdc.gov"
+        ),
+        version = "0.2.0-oas3"
+    )
+)
+@SecurityScheme(
+    name = "primeSecurity",
+    type = SecuritySchemeType.OAUTH2,
+    flows = OAuthFlows(
+        authorizationCode = OAuthFlow(
+            authorizationUrl = "https://hhs-prime.okta.com/oauth/authorize",
+            tokenUrl = "https://hhs-prime.okta.com/oauth/token",
+            scopes = [
+                OAuthScope(
+                    name = "org_admin",
+                    description = "Grants write access to single org"
+                ),
+                OAuthScope(name = "system_admin", description = "Grants access to admin operations"),
```

Comment:
we've usually been calling this "prime_admin" but I think this is fine if you prefer it.

Reply:
> we've usually been calling this "prime_admin" but I think this is fine if you prefer it.

this section is adopted from Michael's PR

Reply:
This is what the existing api docs call it, but agree we should change it.

Reply:
So to be clear, "system_admin" should be changed to "prime_admin" and where you have prime_admin now, on line 153, should get changed back to "org_admin". @mkalish please confirm